<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CoAPClientUi</class>
 <widget class="QWidget" name="CoAPClientUi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>718</width>
    <height>549</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>4</number>
   </property>
   <property name="topMargin">
    <number>4</number>
   </property>
   <property name="rightMargin">
    <number>4</number>
   </property>
   <property name="bottomMargin">
    <number>4</number>
   </property>
   <property name="spacing">
    <number>4</number>
   </property>
   <item row="2" column="1">
    <widget class="QSpinBox" name="spinBoxServerPort"/>
   </item>
   <item row="1" column="1">
    <widget class="QComboBox" name="comboBoxServerIp">
     <property name="editable">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Server IP</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="label_3">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string notr="true"/>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Server Port</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_4">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Protocol</string>
     </property>
    </widget>
   </item>
   <item row="9" column="0" colspan="2">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="spacing">
      <number>4</number>
     </property>
     <item row="3" column="0" colspan="2">
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Token</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonTokenFormat">
         <property name="toolTip">
          <string>Change the token format</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonTokenRemove">
         <property name="toolTip">
          <string>Remove the token from list</string>
         </property>
         <property name="statusTip">
          <string>Token text format</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonTokenAdd">
         <property name="toolTip">
          <string>Add the token to list</string>
         </property>
         <property name="statusTip">
          <string>Save the token</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonTokenEdit">
         <property name="toolTip">
          <string>Edit token list</string>
         </property>
         <property name="statusTip">
          <string>Edit token</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="QTextEdit" name="textEditPayload">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="placeholderText">
        <string>Please input CoAP payload</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0" colspan="2">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="font">
          <font>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Option</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonOptionFormat">
         <property name="toolTip">
          <string>Change the optioin format</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonOptionRemove">
         <property name="toolTip">
          <string>Remove the option from list</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonOptionAdd">
         <property name="toolTip">
          <string>Add the option to list</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonOptionEdit">
         <property name="toolTip">
          <string>Edit option list</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="5" column="0" colspan="2">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label_6">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Resource</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonResourceFormat">
         <property name="toolTip">
          <string>Change the resource format</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonResourceRemove">
         <property name="toolTip">
          <string>Remove the resource from list</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonResourceAdd">
         <property name="toolTip">
          <string>Add the resource to list</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonResourceEdit">
         <property name="toolTip">
          <string>Edit option resource</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="0" column="0" rowspan="2" colspan="2">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="label_8">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <bold>false</bold>
          </font>
         </property>
         <property name="text">
          <string>Payload</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonPayloadFormat">
         <property name="toolTip">
          <string>Payload format</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonPayloadLoad">
         <property name="toolTip">
          <string>Preset payload list</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonPayloadSave">
         <property name="toolTip">
          <string>Save current payload</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QToolButton" name="toolButtonPayloadEdit">
         <property name="toolTip">
          <string>Edit payload list</string>
         </property>
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="9" column="0" colspan="2">
      <layout class="QGridLayout" name="gridLayout_4">
       <property name="spacing">
        <number>4</number>
       </property>
       <item row="2" column="1">
        <widget class="QPushButton" name="pushButtonPatch">
         <property name="text">
          <string notr="true">Patch</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButtonGet">
         <property name="text">
          <string notr="true">Get</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="pushButtonFetch">
         <property name="text">
          <string notr="true">Fetch</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButtonPut">
         <property name="text">
          <string notr="true">Put</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="pushButtonDelete">
         <property name="text">
          <string notr="true">Delete</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButtonPost">
         <property name="text">
          <string notr="true">Post</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QPushButton" name="pushButtonIPatch">
         <property name="text">
          <string notr="true">iPatch</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="pushButtonPresets">
         <property name="text">
          <string>Presets</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="8" column="0" colspan="2">
      <widget class="QComboBox" name="comboBoxOption">
       <property name="editable">
        <bool>true</bool>
       </property>
       <property name="placeholderText">
        <string>Please input CoAP option</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0" colspan="2">
      <widget class="QComboBox" name="comboBoxResource">
       <property name="editable">
        <bool>true</bool>
       </property>
       <property name="placeholderText">
        <string>Please input CoAP resource</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0" colspan="2">
      <widget class="QComboBox" name="comboBoxToken">
       <property name="editable">
        <bool>true</bool>
       </property>
       <property name="placeholderText">
        <string>Please input CoAP token</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="2" rowspan="10">
    <widget class="QWidget" name="widget" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QComboBox" name="comboBoxProtocol"/>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QPushButton" name="pushButtonOpen">
     <property name="text">
      <string>Open Client</string>
     </property>
    </widget>
   </item>
   <item row="5" column="0" colspan="2">
    <widget class="QPushButton" name="pushButtonClose">
     <property name="text">
      <string>Close Client</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Client Settings</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="toolButtonClient">
       <property name="text">
        <string>...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
